lib/workers/repository/package-files.ts

```diff
@@ -53,28 +47,30 @@ export class PackageFiles {
     let header = '';
     let removed = false;
     let truncated = false;
-    let restore: (() => void) | null = null;
+
+    const data = new Map(clone(Array.from(this.data)));
+
+    // filter all deps with skip reason
+    for (const managers of [...data.values()].filter(is.truthy)) {
+      for (const files of Object.values(managers).filter(is.truthy)) {
+        for (const file of files.filter((f) => is.truthy(f.deps))) {
+          file.deps = file.deps.filter(is.truthy).filter((d) => !d.skipReason);
+        }
+      }
+    }
```

Comment:
does the two first `is.truthy` needed? say for example we set a value of `undefiend|null` for some key in the map, is it possible to set a prop on an `undefiend|null`? i.e. `undefined.files = []`?
i know js is quirky but wouldnt this fail on runtime?

Reply:
But this is Object.values(). not Object.keys()?

Reply:
yes, i only get `values` which can be `null` or `undefined`. it's just defense code because  of some misbehaving managers

Reply:
all i meant to say is that say we have some manager that has undefined as its value i.e.
```json
{
  "baseBranch": "main",
  "config": {
    "docker-compose":  undefined | null | empty Array
}
```

if will be filtered out in the next for loop regardless of `is.truthy`
and also i think that it should be `manager` and not `managers`, as we have files per a single manager

```suggestion
    // filter all deps with skip reason
    for (const manager of [...data.values()].filter(is.truthy)) {
      for (const files of Object.values(manager).filter(is.truthy)) {
        for (const file of files.filter((f) => is.truthy(f.deps))) {
          file.deps = file.deps.filter(is.truthy).filter((d) => !d.skipReason);
        }
      }
    }
```